### PR TITLE
[SCSE-86] Initialize admin panel backend in monorepo

### PR DESCRIPTION
**Pull request by limivann**
- Testing are not added yet

Edit: done

### CODE REVIEW

File: apps/admin-be/src/types/ProductType.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export type ProductType = {
+  id: string;
+  name: string;
+  price: number; // Price of the product in cents ($19 = 1900).
+  images: string[]; // URL of the images.
+  sizes: ProductSizeTypes[];
+  productCategory: ProductCategoryType;
+  isAvailable?: boolean;
+};
```
Comment (RealDyllon) [inline apps/admin-be/src/types/ProductType.ts:22]: We should move this to a new lib in `packages/types`
More info here: https://turbo.build/repo/docs/handbook/sharing-code

File: apps/admin-be/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "tsconfig/node.json",
+  "compilerOptions": {
+    "outDir": "dist"
+  },
+  "include": ["src/**/*"]
+}
```
Comment (RealDyllon) [inline apps/admin-be/tsconfig.json:7]: We should be extending the common tsconfig - refer to the other apps to see how we're doing that.

File: apps/admin-be/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    root: true,
+    extends: ["custom"],
+};
```
Comment (RealDyllon) [inline apps/admin-be/.eslintrc.js:4]: good use of the shared custom eslint config

File: apps/admin-be/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[{package.json,*.yml}]
+indent_style = space
+indent_size = 2
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (RealDyllon) [inline apps/admin-be/.editorconfig:16]: should this be committed 🤔

File: apps/admin-be/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "admin-be",
+  "version": "1.0.0",
+  "description": "backend for admin panel",
+  "main": "dist/index.js",
+  "license": "MIT",
+  "scripts": {
+    "devStart": "nodemon --ignore tests/ --watch src -e ts,tsx --exec ts-node src/index.ts",
```
Comment (RealDyllon) [inline apps/admin-be/package.json:8]: lets rename this script to `dev`, so it aligns with the turbo pipeline names (see `turbo.json`).

File: apps/admin-be/package.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    "build-ts": "tsc",
+    "lint": "tsc --noEmit && eslint \"**/*.{js,ts}\" --quiet --fix",
+    "serve": "node dist/index.js",
+    "build": "yarn run lint && yarn run build-ts",
```
Comment (RealDyllon) [inline apps/admin-be/package.json:12]: can remove the lint job here, let's call tsc directly. This makes the `build-ts` script redundant.

File: apps/admin-be/package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "admin-be",
+  "version": "1.0.0",
+  "description": "backend for admin panel",
+  "main": "dist/index.js",
+  "license": "MIT",
```
Comment (RealDyllon) [inline apps/admin-be/package.json:6]: @teamoteh and I need to decide on a license, for now let's remove this field.

File: apps/admin-be/src/controllers/authController.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    res.status(400).json("no first name");
+  }
+  res.sendStatus(201);
+};
+
+const AuthController = {
+  Login,
+};
+
+export { AuthController };
```
Comment (RealDyllon) [inline apps/admin-be/src/controllers/authController.ts:14]: Can we add a comment here that states that this auth is purely boilerplate. 

Later in another PR we will import AWS Cognito.

**Review by RealDyllon — DISMISSED** [review]
Good stuff @limivann, just a few changes that need to be made.

File: apps/admin-be/src/types/ProductType.ts (hunk 1 above)
Comment (limivann) [reply apps/admin-be/src/types/ProductType.ts:22]: Yep, will move.

File: apps/admin-be/.editorconfig (hunk 4 above)
Comment (limivann) [reply apps/admin-be/.editorconfig:16]: This is for vscode users, prettier by default does not format on save

File: apps/admin-be/package.json (hunk 7 above)
Comment (limivann) [reply apps/admin-be/package.json:6]: Alright noted

File: apps/admin-be/tsconfig.json (hunk 2 above)
Comment (limivann) [reply apps/admin-be/tsconfig.json:7]: Line 2 means we are extending from the shared tsconfig no?

**Review by teamoteh — APPROVED** [review]

**Review by teamoteh — APPROVED** [review]

**Pull request merged**